Compile error in MinGW

compiler: g++ 6.3
platform: windows10 / x86_64
range-v3: range-v3-0.2.4

================================

Scanning dependencies of target container_conversion                                                                                
[ 54%] Building CXX object test/CMakeFiles/container_conversion.dir/container_conversion.cpp.obj                                    
In file included from C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/detail/optional.hpp:20:0,                                  
                 from C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/utility/semiregular.hpp:21,                                
                 from C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/utility/basic_iterator.hpp:26,                             
                 from C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/utility/iterator.hpp:28,                                   
                 from C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/begin_end.hpp:27,                                          
                 from C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/core.hpp:17,                                               
                 from C:\Users\JohnChang\range-v3-0.2.4\test\container_conversion.cpp:18:                                           
C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/detail/variant.hpp: In member function 'meta::v1::if_c<(N != M)> ranges::v3::deta
il::construct_fn<N, Ts>::operator()(ranges::v3::detail::indexed_datum<U, std::integral_constant<long long unsigned int, M> >&) [with
 U = meta::v1::nil_; long long unsigned int M = 0ull; long long unsigned int N = 1ull; Ts = {ranges::v3::_counted_iterator_::counted
_iterator<ranges::v3::_basic_iterator_::basic_iterator<ranges::v3::iota_view<int, void> >, void>&&}]':                              
C:/Users/JohnChang/range-v3-0.2.4/include/range/v3/detail/variant.hpp:398:17: error: 'noreturn' function does return [-Werror]      
                 }

oops!
another compile failed
[ 87%] Building CXX object test/utility/CMakeFiles/utility.functional.dir/functional.cpp.obj
In file included from C:/Users/JohnChang/range-v3/include/range/v3/utility/box.hpp:25:0,
                 from C:/Users/JohnChang/range-v3/include/range/v3/utility/functional.hpp:24,
                 from C:\Users\JohnChang\range-v3\test\utility\functional.cpp:14:
C:\Users\JohnChang\range-v3\test\utility\functional.cpp: In function 'int main()':
C:\Users\JohnChang\range-v3\test\utility\functional.cpp:269:47: error: '__vectorcall' was not declared in this scope
         CONCEPT_ASSERT(ranges::Invocable<void(__vectorcall*)()>());

--- a/include/range/v3/detail/config.hpp
+++ b/include/range/v3/detail/config.hpp
@@ -16,9 +16,10 @@
 #define RANGES_V3_DETAIL_CONFIG_HPP
 
 #include <iosfwd>
-#if(defined(NDEBUG) && !defined(RANGES_ENSURE_MSG)) || \
+#if (defined(NDEBUG) && !defined(RANGES_ENSURE_MSG)) || \
     (!defined(NDEBUG) && !defined(RANGES_ASSERT) && \
-     defined(__GNUC__) && !defined(__clang__) && __GNUC__ < 5)
+     defined(__GNUC__) && !defined(__clang__) && \
+     (__GNUC__ < 5 || defined(__MINGW32__)))
 #include <cstdio>
 #include <cstdlib>
 
@@ -40,7 +41,8 @@ namespace ranges
 #endif
 
 #ifndef RANGES_ASSERT
-#if !defined(NDEBUG) && defined(__GNUC__) && !defined(__clang__) && __GNUC__ < 5
+#if !defined(NDEBUG) && defined(__GNUC__) && !defined(__clang__) && \
+    (__GNUC__ < 5 || defined(__MINGW32__))
 #define RANGES_ASSERT(...) \
     static_cast<void>((__VA_ARGS__) ? void(0) : \
         ::ranges::detail::assert_failure(__FILE__, __LINE__, "assertion failed: " #__VA_ARGS__))

